### DISCUSSION TITLE
Build a simple Content Management System (CMS) with ParcelJS

### DISCUSSION THREAD

**Discussion by Doogiemuc** [parcel-bundler/parcel#5780] [Q&A]
Hello everyone!

I have an idea of a really very simple Content Management System that I'd like to build with parcel. I would like to get feedback from the parcel community if my concept could probably work, or if I oversaw something.

## The idea

1. Have some static HTML files. Some divs in the HTML are specially marked as beeing "editable".
2. `parcel serve`  these assets - but with a custom writen webserver/middleware/parcel bunder/parcel plugin, ....
3. The HTML files contain a WYSIWYG editor (e.g. editor.js) that allows the user to edit the content inside the browser.
4. When he saves, the edited content is locally stored to the file system
5. parcel watcher picks up the changes and refreshes the page.

## Thoughts

Of course I know, that the browser can (normally) not directly access the local filesystem.  But a [customized parcel bundler](https://parceljs.org/api.html) could.  (Step 2.)

Editor.js is quite clever. It does not simply outputs the edited HTML, Instead it outputs a custom format. (Actually an AST.) Now this needs to be formated to the locally stored format, which for example could be Markdown.  Editor.js plugins - that I would implement - can do that.

## Questions

- Do you think this could work?
- Did I oversee something? Is there any blocker?

**Comment by mischnic** [comment]
For Parcel 2, this sounds possible.

- A reporter that opens up a server and writes to the fs.
- A runtime or transformer that adds code into JS/HTML (depends on how exactly this should work)

**Reply by Doogiemuc** [reply]
Thank you very much for the quich response. Ok then I am desperately looking forward to Parcel 2 - (I assume as everyone else, too:-)   Keep up the good work.  I really love Parcel for its simplicity (compared to other "web packing tools" :-)      I'll come back here, if I succeed with my idea.
